oak_functions/loader/benches/lookup.rs

```diff
@@ -33,125 +31,199 @@ use oak_functions_loader::{
 use prost::Message;
 use proto::{benchmark_request::Action, BenchmarkRequest, LookupTest};
 use rand::SeedableRng;
-use std::{collections::HashMap, sync::Arc, time::Duration};
-use test::Bencher;
+use std::{collections::HashMap, sync::Arc};
+use tokio::runtime::Runtime;
 
 const MANIFEST_PATH: &str = "examples/benchmark/module/Cargo.toml";
+const SINGLE_REQUEST_ITERATIONS: u32 = 1;
+const MULTI_REQUEST_ITERATIONS: u32 = 101;
 
-#[bench]
-fn small_fixed_data_single_lookup(bencher: &mut Bencher) {
-    let entries = hashmap! {
-        b"key_0".to_vec() => br#"value_0"#.to_vec(),
-        b"key_1".to_vec() => br#"value_1"#.to_vec(),
-        b"key_2".to_vec() => br#"value_2"#.to_vec(),
-    };
-
-    let test_data = TestData {
-        test_key: br#"key_1"#.to_vec(),
-        expected_value: br#"value_1"#.to_vec(),
-        entries,
-    };
-
-    run_lookup_bench(bencher, test_data, 1, Duration::from_millis(5));
-}
-
-#[bench]
-fn small_fixed_data_multi_lookup(bencher: &mut Bencher) {
-    let entries = hashmap! {
-        b"key_0".to_vec() => br#"value_0"#.to_vec(),
-        b"key_1".to_vec() => br#"value_1"#.to_vec(),
-        b"key_2".to_vec() => br#"value_2"#.to_vec(),
-    };
-
-    let test_data = TestData {
-        test_key: br#"key_1"#.to_vec(),
-        expected_value: br#"value_1"#.to_vec(),
-        entries,
-    };
+fn key_size(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+    let wasm_module_bytes = bulid_wasm_module(MANIFEST_PATH);
 
-    run_lookup_bench(bencher, test_data, 101, Duration::from_millis(10));
-}
+    let mut group = c.benchmark_group("key_size");
+    let plot_config = PlotConfiguration::default().summary_scale(AxisScale::Logarithmic);
+    group.plot_config(plot_config);
+    for size in [16usize, 128usize, 1_024usize].iter() {
```

Comment:
If you use `into_iter` then I think you get actual values, not refs. Also, I thought `into_iter` is actually the default if you don't specify anything, but I may be wrong.

Reply:
Done.